apps/etterlatte-behandling/src/main/kotlin/behandling/BehandlingFactory.kt

```diff
@@ -14,40 +14,82 @@ import no.nav.etterlatte.libs.common.behandling.BehandlingType
 import no.nav.etterlatte.libs.common.behandling.Persongalleri
 import no.nav.etterlatte.libs.common.behandling.RevurderingAarsak
 import no.nav.etterlatte.libs.common.behandling.SakType
+import no.nav.etterlatte.libs.common.grunnlag.Grunnlagsopplysning
+import no.nav.etterlatte.libs.common.grunnlag.NyeSaksopplysninger
+import no.nav.etterlatte.libs.common.grunnlag.opplysningstyper.Opplysningstype
+import no.nav.etterlatte.libs.common.grunnlag.opplysningstyper.SoeknadMottattDato
+import no.nav.etterlatte.libs.common.gyldigSoeknad.GyldighetsResultat
+import no.nav.etterlatte.libs.common.gyldigSoeknad.VurderingsResultat
 import no.nav.etterlatte.libs.common.person.Folkeregisteridentifikator
 import no.nav.etterlatte.libs.common.sak.Sak
+import no.nav.etterlatte.libs.common.tidspunkt.Tidspunkt
+import no.nav.etterlatte.libs.common.tidspunkt.toLocalDatetimeUTC
+import no.nav.etterlatte.libs.common.tidspunkt.toTidspunkt
+import no.nav.etterlatte.libs.common.toJsonNode
 import no.nav.etterlatte.oppgaveny.OppgaveServiceNy
-import no.nav.etterlatte.sak.SakDao
-import no.nav.etterlatte.sak.filterSakerForEnheter
+import no.nav.etterlatte.sak.SakService
 import org.slf4j.LoggerFactory
 import java.time.LocalDateTime
 
 class BehandlingFactory(
     private val oppgaveService: OppgaveServiceNy,
     private val grunnlagService: GrunnlagService,
     private val revurderingService: RevurderingService,
-    private val sakDao: SakDao,
+    private val gyldighetsproevingService: GyldighetsproevingService,
+    private val sakService: SakService,
     private val behandlingDao: BehandlingDao,
     private val hendelseDao: HendelseDao,
     private val behandlingHendelser: BehandlingHendelserKafkaProducer,
     private val featureToggleService: FeatureToggleService
 ) {
     private val logger = LoggerFactory.getLogger(this::class.java)
+
+    fun opprettBehandling(behov: NyBehandlingRequest): Behandling {
+        val soeker = behov.persongalleri.soeker
+
+        val sak = sakService.finnEllerOpprettSak(soeker, behov.sakType)
+
+        val behandling = opprettBehandling(
+            sak.id,
+            behov.persongalleri,
+            behov.mottattDato,
+            Vedtaksloesning.GJENNY
+        )
+
+        if (behandling == null) {
+            throw IllegalStateException("Kunne ikke opprette behandling")
+        }
+
+        val gyldighetsvurdering = GyldighetsResultat(
+            VurderingsResultat.KAN_IKKE_VURDERE_PGA_MANGLENDE_OPPLYSNING,
+            emptyList(),
+            Tidspunkt.now().toLocalDatetimeUTC()
+        )
+
+        gyldighetsproevingService.lagreGyldighetsproeving(behandling.id, gyldighetsvurdering)
+
+        val mottattDato = LocalDateTime.parse(behov.mottattDato)
+        val kilde = Grunnlagsopplysning.Privatperson(soeker, mottattDato.toTidspunkt())
+
+        val opplysninger = listOf(
+            lagOpplysning(Opplysningstype.SPRAAK, kilde, behov.spraak.toJsonNode()),
```

Comment:
Vi må få lagt på opplysningstypen `INNSENDER_SOEKNAD_V1` også da den brukes i brev.

Reply:
Lurer litt på om det kanskje er mer hensiktsmessig å opprette en ny type for innsender: `INNSENDER_PDL_V1` 

På den måten er det mindre for sb å fylle ut _og_ vi sikrer korrekte data uten skrivefeil fra PDL, som for resten av persongalleriet. 

Hva tror du?

Reply:
Kan vi ikke bare bruke innsender typen som du sender inn fra fronten da og smelle den inn i grunnlag?

Reply:
Etter mye frem og tilbake ser det ut til at det å justere bruken av innsender på brevsiden er det mest fornuftige. Vi bruker uansett bare fnr. til innsender, så det er ingen vits å hente hele innsender-objektet.